Makefile

```diff
@@ -0,0 +1,98 @@
+# Makefile
+
+GO_PREFIX ?= github.com/virtual-vgo/vvgo
+
+# Tests
+
+.PHONY: fmt vet test
+fmt:
+	gofmt -d .
+
+vet:
+	go vet $(GO_PREFIX)/...
+
+TEST_FLAGS ?= -race
+test: fmt vet
+	go test $(TEST_FLAGS) $(GO_PREFIX)/...
+
+# Build vvgo
+
+# Use go build tools caching, so mark this as a phony target
+.PHONY: vvgo
+BIN_PATH ?= .
+BUILD_FLAGS ?= -v
+vvgo:
+	go generate ./... && go build -v -o $(BIN_PATH)/vvgo $(GO_PREFIX)/cmd/vvgo
+
+# Build images
+
+VVGO_IMAGE_NAME=vvgo
+PAGE_CACHE_IMAGE_NAME=page-cache
+OBJECT_CACHE_IMAGE_NAME=object-cache
+
+GITHUB_REPO ?= virtual-vgo/vvgo
+IMAGE_REPO ?= docker.pkg.github.com/$(GITHUB_REPO)
+VVGO_IMAGE_CACHE ?= $(IMAGE_REPO)/$(VVGO_IMAGE_NAME):latest
+PAGE_CACHE_IMAGE_CACHE ?= $(IMAGE_REPO)/$(PAGE_CACHE_IMAGE_NAME):latest
+OBJECT_CACHE_IMAGE_CACHE ?= $(IMAGE_REPO)/$(OBJECT_CACHE_IMAGE_NAME):latest
+
+.PHONY: images images/vvgo images/object-cache images/page-cache
+
+images/vvgo-builder:
+	docker pull $(VVGO_IMAGE_CACHE)-builder || true
+	docker build . \
+		--file Dockerfile \
+		--target builder \
+		--cache-from=$(VVGO_IMAGE_CACHE)-builder \
+		--build-arg GITHUB_SHA=$GITHUB_SHA \
+		--build-arg GITHUB_REF=$GITHUB_REF \
+		--tag builder
+
+images/vvgo: images/vvgo-builder
+	docker pull $(VVGO_IMAGE_CACHE) || true
+	docker build . \
+		--file Dockerfile \
+		--target vvgo \
+		--cache-from=$(VVGO_IMAGE_CACHE)-builder \
+		--cache-from=$(VVGO_IMAGE_CACHE) \
+		--tag artifact
+
+images/page-cache:
```

Comment:
IMHO we should really rename this to ingress or (reverse?) proxy. Maybe later, though.